packages/shared/auth0-client/src/index.ts

```diff
@@ -177,6 +192,37 @@ export default class Auth0Client {
     });
   }
 
+  async searchUsers(page: number, pageSize: number, sort: string, sortDir: number, query: string): Promise<APIResponse<Auth0SearchResults>> {
+    return this.getMachineToMachineInstance().then(instance => {
+      return instance.get('/users', {
+        params: {
+          page: page,
+          per_page: pageSize,
+          include_totals: true,
+          sort: Auth0SearchSortFields.get(sort) + ':' + sortDir,
+          connection: 'Sierra-Connection',
+          q: query.split(' ').map(token => 'name:*' + token + '* OR email:*' + token + '*').join(' '),
```

Comment:
Can we break out this query generation to a separate function so we can unit test that the correct Lucene query is generated for a given set of search params?

Reply:
Good shout, done.